src/cli/src/operations.c

```diff
@@ -666,13 +666,17 @@ char *talk_operation(char *tokens[TOKEN_LIST_SIZE], chiventure_ctx_t *ctx)
     {
         return "No one by that name wants to talk.";
     }
-
-    if (npc->dialogue == NULL)
+    
+    quest_ctx_t qctx;
+    quest_ctx_init(&qctx, ctx->game->curr_player, ctx->game->all_quests);
```

Comment:
this is going to immediately fail the `assert(quest_ctx != NULL);` statement in the first line of `quest_ctx_init()` since `quest_ctx_t qctx;` is NULL-initialized by default, i think you meant to do this:
```suggestion
    quest_ctx_t qctx = quest_ctx_new(ctx->game->curr_player,  ctx->game->all_quests);
```
and then free this `qctx` at some later point in this function ofc

Reply:
Yes, nice catch!

Reply:
in my suggested change, I forgot to make the new `quest_ctx_t *qctx` a pointer, so I just wanted to make another comment to make sure you didn't make the same mistake

Reply:
Fixed I think